app/js/lib/foxbox/network.js

```diff
@@ -0,0 +1,206 @@
+
+    this[p.local] = false;
+    this[p.remote] = false;
+    this[p.pingInterval] = null;
+  }
+
+  /**
+   * Attach event listeners related to the connection status.
+   *
+   * @return {Promise}
+   */
+  init() {
+    const pingBox = this[p.pingBox].bind(this);
+
+    window.addEventListener('online', pingBox);
+    window.addEventListener('offline', pingBox);
+
+    if ('connection' in navigator && 'onchange' in navigator.connection) {
+      navigator.connection.addEventListener('change', pingBox);
+
+      // We also ping the box every few minutes to make sure it's still there.
+      this[p.pingInterval] = setInterval(pingBox,
+        this.settings.onlineCheckingLongInterval);
+    } else {
+      // If the Network Information API is not implemented, fallback to polling.
+      this[p.pingInterval] = setInterval(pingBox,
+        this.settings.onlineCheckingInterval);
+    }
+
+    this[p.pingBox]();
+
+    return Promise.resolve();
+  }
+
+  get origin() {
+    if (this[p.local]) {
+      return this.localOrigin;
+    } else if (this[p.remote]) {
+      return this.tunnelOrigin;
+    }
+
+    console.error('The box is out of reach.');
+    return this.localOrigin;
+  }
+
+  get localOrigin() {
+    return `${this.settings.localScheme}://${this.settings.localHostname}:${this.settings.localPort}`;
+  }
+
+  get tunnelOrigin() {
+    return `${this.settings.tunnelScheme}://${this.settings.tunnelHostname}:${this.settings.tunnelPort}`;
+  }
+
+  get connected() {
+    return this[p.local] || this[p.remote];
+  }
+
+  get connection() {
+    if (this[p.local]) {
+      return 'local';
+    } else if (this[p.remote]) {
+      return 'remote';
+    }
+    return 'unknown';
+  }
+
+  /**
+   * Request a JSON from a specified URL.
+   *
+   * @param {string} url The URL to send the request to.
+   * @param {string} method The HTTP method (defaults to "GET").
+   * @param {Object} body An object of key/value.
+   * @return {Promise}
+   */
+  fetchJSON(url, method = 'GET', body = undefined) {
+    return this[p.fetch](url, 'application/json', method, body)
+      .then(response => response.json());
+  }
+
+  /**
+   * Request a Blob from a specified URL.
+   *
+   * @param {string} url The URL to send the request to.
+   * @param {string} blobType The Blob mime type (eg. image/jpeg).
+   * @param {string} method The HTTP method (defaults to "GET").
+   * @param {Object} body An object of key/value.
+   * @return {Promise<Blob>}
+   */
+  fetchBlob(url, blobType, method, body) {
+    return this[p.fetch](url, blobType, method, body)
+      .then(response => response.blob());
+  }
+
+  /**
+   * Request a content of the specified type from a specified URL.
+   *
+   * @todo Detect if the URL is relative, if so prepend this.origin.
+   *
+   * @param {string} url The URL to send the request to.
+   * @param {string} accept The content mime type (eg. image/jpeg).
+   * @param {string=} method The HTTP method (defaults to "GET").
```

Comment:
Well, my main rationale is that JSDoc is tool/compiler/framework agnostic while Google Closure annotations are not. And even if start using Google Closure Compiler, it should not affect the way we document code, JSDoc and document generation tool based on it evolutionize independently and rapidly :)